lib/ledger_sync/adaptors/quickbooks_online/adaptor.rb

```diff
@@ -54,60 +59,73 @@ def authorization_url(redirect_uri:)
           oauth_client.authorization_url(redirect_uri: redirect_uri)
         end
 
-        def find(resource:, id:)
-          resource = resource.to_s
-          url = "#{oauth_base_uri}/#{resourcify(resource)}/#{id}"
+        def find(path:)
+          url = "#{oauth_base_uri}/#{path}"
 
-          response = request(:get, url, headers: OAUTH_HEADERS.dup)
-          JSON.parse(response.body).dig(resource.classify)
+          request(
+            headers: OAUTH_HEADERS.dup,
+            method: :get,
+            url: url
+          )
         end
 
-        def parse_operation_error(error:, operation:)
-          return nil unless error.is_a?(OAuth2::Error)
+        def oauth
+          OAuth2::AccessToken.new(
+            oauth_client.client,
+            access_token,
+            refresh_token: refresh_token
+          )
+        end
 
-          Util::OperationErrorParser.new(
-            error: error,
-            operation: operation
-          ).parse
+        def oauth_client
+          @oauth_client ||= OAuthClient.new(
+            client_id: client_id,
+            client_secret: client_secret
+          )
         end
 
-        def post(resource:, payload:)
-          resource = resource.to_s
-          url = "#{oauth_base_uri}/#{resourcify(resource)}"
+        def post(path:, payload:)
+          url = "#{oauth_base_uri}/#{path}"
 
-          response = request(
-            :post,
-            url,
+          request(
             headers: OAUTH_HEADERS.dup,
-            body: payload.to_json
+            method: :post,
+            body: payload.to_json,
+            url: url
           )
-          JSON.parse(response.body).dig(resource.classify)
         end
 
-        def query(resource:, query:, limit: 10, offset: 1)
-          resource = resource.to_s
-          full_query = "SELECT * FROM #{resource.classify} WHERE #{query} STARTPOSITION #{offset} MAXRESULTS #{limit}"
+        def query(limit: 10, offset: 1, query:, resource_class:)
```

Comment:
@ryanwjackson I'm getting an error related to this method when I run things locally. The QB accounts searcher `adaptors/quickbooks_online/account/searcher.rb` calls this `query` method but still passes in a `resource` param instead of the newly renamed `resource_class` param. Perhaps a lot of the searchers need to be updated too?